src/lib/alloc.c

```diff
@@ -647,14 +647,22 @@ static void alloc_trace_heap(enum mem_zone zone, uint32_t caps, size_t bytes)
 
 #define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
 	do { \
-		if (!ptr) { \
-			tr_err(&mem_tr, "failed to alloc 0x%x bytes zone 0x%x caps 0x%x flags 0x%x", \
+		if (!(ptr)) { \
+			tr_err(&mem_tr, \
+			       "failed to alloc 0x%x bytes zone 0x%x caps 0x%x flags 0x%x", \
 			       bytes, zone, caps, flags); \
 			alloc_trace_heap(zone, caps, bytes); \
 		} \
 	} while (0)
 #else
-#define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags)
+#define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
+	do { \
+		if (!(ptr)) { \
+			tr_err(&mem_tr, \
+			       "failed to alloc 0x%x bytes zone 0x%x caps 0x%x flags 0x%x", \
+			       bytes, zone, caps, flags); \
+		} \
```

Comment:
Sorry, I was meaning from my comment on the previous PR it would be good to dump the heap blocks on any allocation error. I see we are already doing this but in debug mode only ?

Reply:
> Sorry, I was meaning from my comment on the previous PR it would be good to dump the heap blocks on any allocation error. I see we are already doing this but in debug mode only ?

That's true. One more bad thing is that our trace are always filtered out by the trace filtering, that's why I tried to set the filtering to N by default

Reply:
Can we do something like this (it sames duplicating the print).

```c
#define _DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
	do { \
		if (!(ptr)) { \
			tr_err(&mem_tr, \
			       "failed to alloc 0x%x bytes zone 0x%x caps 0x%x flags 0x%x", \
			       bytes, zone, caps, flags); \
		} \
	} while (0)

#if DEBUG
#define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
                       _DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
			alloc_trace_heap(zone, caps, bytes);
#else
#define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
                       _DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags)
#endif
```

Reply:
it doesnt look like this feedback was addressed?